packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,349 @@
+    double angle = 0.0,
+    Anchor anchor = Anchor.topLeft,
+    int? priority,
+  })  : _transform = Transform2D(),
+        _anchor = anchor,
+        _size = NotifyingVector2.copy(size ?? Vector2.zero()),
+        _scaledSize = Vector2.zero(),
+        super(priority: priority) {
+    if (position != null) {
+      _transform.position = position;
     }
-    return _scaledSizeCache.value!;
+    if (angle != 0) {
+      _transform.angle = angle;
+    }
+    if (scale != null) {
+      _transform.scale = scale;
+    }
+    _size.addListener(_onModifiedSizeOrAnchor);
+    _size.addListener(_onModifiedSizeOrScale);
+    _transform.scale.addListener(_onModifiedSizeOrScale);
+    _onModifiedSizeOrAnchor();
+    _onModifiedSizeOrScale();
   }
 
-  /// Get the absolute position, with the anchor taken into consideration
-  Vector2 get absolutePosition => absoluteParentPosition + position;
+  /// The total transformation matrix for the component. This matrix combines
+  /// translation, rotation and scale transforms into a single entity. The
+  /// matrix is cached and gets recalculated only as necessary.
+  Matrix4 get transformMatrix => _transform.transformMatrix;
+
+  /// The position of this component's anchor on the screen.
+  NotifyingVector2 get position => _transform.position;
+  set position(Vector2 position) => _transform.position = position;
+
+  /// X position of this component's anchor on the screen.
+  double get x => _transform.x;
+  set x(double x) => _transform.x = x;
+
+  /// Y position of this component's anchor on the screen.
+  double get y => _transform.y;
+  set y(double y) => _transform.y = y;
+
+  /// Rotation angle (in radians) of the component. The component will be
+  /// rotated around its anchor point in the clockwise direction if the
+  /// angle is positive, or counterclockwise if the angle is negative.
+  double get angle => _transform.angle;
+  set angle(double a) => _transform.angle = a;
+
+  /// The scale factor of this component. The scale can be different along
+  /// the X and Y dimensions. A scale greater than 1 makes the component
+  /// bigger, and less than 1 smaller. The scale can also be negative,
+  /// which results in a mirror reflection along the corresponding axis.
+  NotifyingVector2 get scale => _transform.scale;
+  set scale(Vector2 scale) => _transform.scale = scale;
+
+  /// Anchor point for this component. An anchor point describes a point
+  /// within the rectangle of size [size]. This point is considered to
+  /// be the logical "center" of the component. This can be visualized
+  /// as the point where Flame "grabs" the component. All transforms
+  /// occur around this point: the [position] is where the anchor point
+  /// will end up after the component is translated; the rotation and
+  /// scaling also happen around this anchor point.
+  Anchor _anchor;
+  Anchor get anchor => _anchor;
+  set anchor(Anchor anchor) {
+    _anchor = anchor;
+    _onModifiedSizeOrAnchor();
+  }
```

Comment:
a good thing to explain on the docs is what happens when you change the anchor. either
1. the position of the anchor on world space is constant and the component moves to fit the new anchor local position
2. the position of the component on world space is constant and the anchor moves to fit the new local relative position

I don't see clearly which of those is the case by reading the comments on anchor, should probably specify for posterity (as both would make sense to me from different point of views)

Reply:
Good point. I'll add some clarifying note here.